pkg/storer/reserve.go

```diff
@@ -253,18 +255,18 @@ func (db *DB) ReservePutter() storage.Putter {
 }
 
 // EvictBatch evicts all chunks belonging to a batch from the reserve.
-func (db *DB) EvictBatch(ctx context.Context, batchID []byte) (err error) {
+func (db *DB) EvictBatch(ctx context.Context, batchID []byte) error {
 	if db.reserve == nil {
 		// if reserve is not configured, do nothing
 		return nil
 	}
 
-	db.lock.Lock(expiredBatchAccess)
-	db.expiredBatches = append(db.expiredBatches, batchID)
-	db.lock.Unlock(expiredBatchAccess)
+	err := db.batchstore.SaveExpired(batchID)
```

Comment:
So I dont think this makes much sense. It seems to be better to store the expired batches in storer itself. Saving it in batchstore context on callback from batchstore seems like unnecessary work.

You can use the store in storer itself to store it when you get the call and then trigger the eviction. When you are evicting you iterate on the store and get the batches instead of getting them from in-mem. Also, on initialization you will have to check if there are entries left to be processed and trigger the eviction again.